Sources/MediaViewer/UI/PreviewController.swift

```diff
@@ -241,3 +264,26 @@ fileprivate final class Toolbar: UIToolbar {
     // TODO: self-resizing height
     // TODO: transparency background when has any item
 }
+
+extension UIInterfaceOrientation {
+    var mask: UIInterfaceOrientationMask {
+        switch self {
+        case .portrait:
+            return .portrait
+        case .landscapeLeft:
+            return .landscapeLeft
+        case .landscapeRight:
+            return .landscapeRight
+        case .portraitUpsideDown:
+            return .portraitUpsideDown
+        default:
+            return .all
+        }
+    }
+}
+
+internal extension UIViewController {
+    var isDarkMode: Bool {
```

Comment:
In the main app we are separating appDarkMode from deviceDarkMode. Did you check that dark mode is working if I change that setting both from the app settings / device settings?

Reply:
yes, it does (the check is against the traitCollection, which gets set either from window.overrideUserInterfaceStyle, or from the device's dark mode settings. either way, it's up to date.